From 9ae3c6f017f42390e81e9ed194ca6babd1ea85a3 Mon Sep 17 00:00:00 2001
From: Adarsh Kumar <adarsh.kumar@axiado.com>
Date: Fri, 13 Feb 2026 05:54:55 -0800
Subject: [PATCH] Ensure post code manager starts after lpcsnoop

Upstream-Status: Pending

diff --git a/service_files/xyz.openbmc_project.State.Boot.PostCode@.service b/service_files/xyz.openbmc_project.State.Boot.PostCode@.service
index d4e8274..6d77548 100644
--- a/service_files/xyz.openbmc_project.State.Boot.PostCode@.service
+++ b/service_files/xyz.openbmc_project.State.Boot.PostCode@.service
@@ -1,7 +1,11 @@
 [Unit]
 Description=Post code manager (host %i)
+Requires=lpcsnoop.service
+After=lpcsnoop.service
 
 [Service]
+Restart=always
+RestartSec=5
 ExecStart=/usr/bin/env post-code-manager --host %i --config /usr/share/phosphor-post-code-manager/post-code-handlers.json
 SyslogIdentifier=post-code-manager%i
 Type=dbus
-- 
2.34.1

