From ce7e8a66664c02a6f235fa0582cd1adb11136351 Mon Sep 17 00:00:00 2001
From: Adarsh Kumar <adarsh.kumar@axiado.com>
Date: Fri, 13 Feb 2026 04:21:50 -0800
Subject: [PATCH] Remove Aspeed ABR function

Upstream-Status: Pending

diff --git a/bmc/obmc-flash-bmc b/bmc/obmc-flash-bmc
index f9cc4e2..9eec6a2 100644
--- a/bmc/obmc-flash-bmc
+++ b/bmc/obmc-flash-bmc
@@ -646,6 +646,8 @@ function mmc_setprimary() {
 }
 
 function mmc_mirroruboot() {
+    return 0
+
     # Get current boot device; 0-primary_bootdev device; 1 - alt_bootdev
     bootdev=$(cat /sys/kernel/debug/aspeed/sbc/abr_image)
     if [[ "${bootdev}" == "0" ]]; then
-- 
2.34.1

