From 031225459f708a961e47a73f8abbafbf37fe969e Mon Sep 17 00:00:00 2001
From: Ginu George <axiado-2551@axiado.com>
Date: Mon, 13 Apr 2026 05:44:54 -0700
Subject: [PATCH 2/2] Remove Aspeed ABR function

Signed-off-by: Ginu George <axiado-2551@axiado.com>
---
 obmc-flash-bmc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/obmc-flash-bmc b/obmc-flash-bmc
index 18f915f..e4a8b12 100644
--- a/obmc-flash-bmc
+++ b/obmc-flash-bmc
@@ -628,6 +628,8 @@ function mmc_setprimary() {
 }
 
 function mmc_mirroruboot() {
+    return 0
+
     # Get current boot device; 0-primary_bootdev device; 1 - alt_bootdev
     bootdev=$(cat /sys/kernel/debug/aspeed/sbc/abr_image)
     if [[ "${bootdev}" == "0" ]]; then
-- 
2.34.1

